### PR TITLE
Rework on the listener

### PR DESCRIPTION
**Pull request by Asakiz**
Also added documentation for the listener.

Signed-off-by: asakiz <asakizin@gmail.com>

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -3,8 +3,7 @@ name: CI
 on:
   push:
     branches:
-      - master
-      - topic/test_ci
+      - v2
```
Comment (otavio) [inline .github/workflows/ci.yml:6]: Split in a specific commit.

**Review by otavio — CHANGES_REQUESTED** [review]
Please organize the commits so we can go for merging it.

**Review by otavio — APPROVED** [review]

**Comment by otavio** [comment]
What else is need for this to be ready for merge?

**Comment by Asakiz** [comment]
For now, nothing. Is ready to merge.

**Pull request merged**